File: services/shop-backend/src/main/java/io/miragon/shop/application/port/inbound/RemoveFromCartUseCase.java
package io.miragon.shop.application.port.inbound;

import io.miragon.shop.domain.article.ArticleId;
import io.miragon.shop.domain.cart.Cart;
import io.miragon.shop.domain.shared.UserId;

public interface RemoveFromCartUseCase {
    Cart removeFromCart(UserId userId, ArticleId articleId);
}

File: services/shop/shop-backend/src/main/kotlin/io/miragon/shop/application/port/inbound/RemoveFromCartUseCase.kt
package io.miragon.shop.application.port.inbound

import io.miragon.shop.domain.article.ArticleId
import io.miragon.shop.domain.cart.Cart
import io.miragon.shop.domain.shared.UserId

interface RemoveFromCartUseCase {
    fun removeFromCart(userId: UserId, articleId: ArticleId): Cart
}